transport/core/config.go

```diff
@@ -0,0 +1,45 @@
+package core
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+)
+
+// TLSClientAuthClientConfig returns a new client authentication TLS
+// configuration that can be used for a client using client auth
+// connecting to the named host.
+func TLSClientAuthClientConfig(cert tls.Certificate, host string) *tls.Config {
+	return &tls.Config{
+		Certificates: []tls.Certificate{cert},
+		RootCAs:      SystemRoots,
+		ServerName:   host,
+		CipherSuites: CipherSuites,
+		MinVersion:   tls.VersionTLS12,
+	}
+}
+
+// TLSClientAuthServerConfig returns a new client authentication TLS
+// configuration for servers expecting mutually authenticated
+// clients. The clientAuth parameter should contain the root pool used
+// to authenticate clients.
+func TLSClientAuthServerConfig(cert tls.Certificate, clientAuth *x509.CertPool) *tls.Config {
+	return &tls.Config{
+		Certificates: []tls.Certificate{cert},
+		RootCAs:      SystemRoots,
+		ClientCAs:    clientAuth,
```

Comment:
It would be nice for this list to be automatically created from a list of certificate providers that are known to issue certificates for trusted clients.